starter-code/ShoppingListWithDetailView/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistdetailview/MainActivity.java

```diff
@@ -20,16 +17,34 @@ protected void onCreate(Bundle savedInstanceState) {
         DBAssetHelper dbSetup = new DBAssetHelper(MainActivity.this);
         dbSetup.getReadableDatabase();
 
-        //Setup the RecyclerView
-        RecyclerView shoppingListRecyclerView = (RecyclerView) findViewById(R.id.shopping_list_recyclerview);
-
-        ShoppingSQLiteOpenHelper db = ShoppingSQLiteOpenHelper.getInstance(this);
-        List<ShoppingItem> shoppingList = db.getShoppingList();
+        Fragment firstFragment = FirstFragment.newInstance(null,this);
 
-        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this,LinearLayoutManager.VERTICAL,false);
-
-        shoppingListRecyclerView.setLayoutManager(linearLayoutManager);
-        shoppingListRecyclerView.setAdapter(new ShoppingListAdapter(shoppingList));
+        //Populate the list of planets
+        getSupportFragmentManager()
+                .beginTransaction()
+                .add(R.id.fragment_container,firstFragment)
+                .commit();
+    }
 
+    @Override
+    public void onItemSelected(String name, String description, String price, String type) {
+        //Replace list fragment with detail fragment
+        //Pass selectedPlanet string to detail fragment in a bundle.
+
+        /**
+         * DEĞİŞTİR!
+         */
+
+        Bundle bundle = new Bundle();
+        bundle.putString("name", name);
+        bundle.putString("description", description);
+        bundle.putString("price", price);
+        bundle.putString("type", type);
```

Comment:
One of the requirements was "Gather the details for the selected item from the database (not passed through the intent)". Technically, you're not passing these values via intent, but it's almost the same thing. This works OK in this case, but what if the item had 100 fields of data? Then you'd want to pass just the item ID here, and make a separate database query from the detail fragment to get the one item you need.